config/config.go

```diff
@@ -5,67 +5,79 @@ import (
 	"errors"
 	"fmt"
 	"os"
-	"reflect"
-	"strconv"
 	"sync/atomic"
 
+	log "github.com/sirupsen/logrus"
+
+	"github.com/kelseyhightower/envconfig"
+
 	"github.com/frain-dev/convoy/config/algo"
 )
 
 var cfgSingleton atomic.Value
 
 type DatabaseConfiguration struct {
-	Dsn string `json:"dsn"`
+	Dsn string `json:"dsn" envconfig:"CONVOY_MONGO_DSN"`
 }
 
 type SentryConfiguration struct {
-	Dsn string `json:"dsn"`
+	Dsn string `json:"dsn" envconfig:"CONVOY_SENTRY_DSN"`
 }
 
 type ServerConfiguration struct {
-	HTTP struct {
-		SSL         bool   `json:"ssl"`
-		SSLCertFile string `json:"ssl_cert_file"`
-		SSLKeyFile  string `json:"ssl_key_file"`
-		Port        uint32 `json:"port"`
-	} `json:"http"`
+	HTTP HTTPServerConfiguration `json:"http"`
+}
+
+type HTTPServerConfiguration struct {
+	SSL         bool   `json:"ssl" envconfig:"SSL"`
+	SSLCertFile string `json:"ssl_cert_file" envconfig:"CONVOY_SSL_CERT_FILE"`
+	SSLKeyFile  string `json:"ssl_key_file" envconfig:"CONVOY_SSL_KEY_FILE"`
+	Port        uint32 `json:"port" envconfig:"PORT"`
 }
 
 type QueueConfiguration struct {
-	Type  QueueProvider `json:"type"`
-	Redis struct {
-		DSN string `json:"dsn"`
-	} `json:"redis"`
+	Type  QueueProvider           `json:"type" envconfig:"CONVOY_QUEUE_PROVIDER"`
+	Redis RedisQueueConfiguration `json:"redis"`
+}
+
```

Comment:
@Subomi we should probably change this to :
`
type QueueConfiguration struct {
	Type  QueueProvider           `json:"type" envconfig:"CONVOY_QUEUE_PROVIDER"`
	Dsn string `json:"dsn"`
       // other details, we can switch based on the value of Type and set it up accordingly 
}
`
I suggest the same for StrategyConfiguration